pkgs/native_toolchain_c/lib/src/cbuilder/compiler_resolver.dart

```diff
@@ -49,7 +49,7 @@ class CompilerResolver {
       return result;
     }
 
-    final targetOS = hookConfig.targetOS;
+    final targetOS = codeConfig.targetOS;
```

Comment:
The uglyness of passing two separate objects around (`HookConfig` and `CodeConfig`) may now go away.

Is the `CompilerResolver.hookConfig` used for anything now that `targetOS` is accessible via the `CodeConfig`?

If not, we can remove the field. The same may also apply to some other classes in the toolchain_c package.